### ISSUE TITLE
Feat: Provide default security context for all Kratix related pods to use

### ISSUE THREAD

**Issue by abangser** [syntasso/kratix#227]
Users require the ability to set a `securityContext` for any workload running in their cluster which means some Kratix managed jobs/pods/containers must adhere to this context.

At a minimum, we need a way for users to declare their default `securityContext`. We may choose to provide overrides for certain workflow containers but that can be put into a separate piece of work if that makes more sense to do.

When working on this, be aware that Kratix runs a number of workloads/containers including (but maybe not limited to):
* Controller manager
* Reader in workflows
* Status writer in workflows
* Work writer in workflows
* Workflow Jobs -> Pods

These should all be solved by setting [`securityContext`](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/) on the pods we own. We may then choose to provide `securityContext` on the pipeline containers so that people can override as they want.

**Comment by abangser** [comment]
Baseline security context per https://kubernetes.io/docs/concepts/security/pod-security-standards/#baseline:

```
                securityContext:
                    capabilities:
                      drop:
                      - ALL
                    privileged: false
                    runAsNonRoot: true
                    seccompProfile:
                      type: RuntimeDefault
```

**Comment by aclevername** [comment]
Work left:
- Use the defaultSecurityContext for the user provided pods
- Allow the setting of the value for the defaultSecruityContext
  - If the user sets one in the Promise, this should take priority over the globally set defaultSecurityContext

**Comment by shano** [comment]
- Fixed issue with defaultSecuriy contrxts
- Need to write tests: 1. to test the logic and 2. to ensure the overrides are working.

**Comment by aclevername** [comment]
Closed by #229
